pages/authPassword/authPassword.js

```diff
@@ -11,14 +11,12 @@ router.get('/', function (req, res) {
 });
 
 router.post('/', function (req, res) {
-  if (req.body.__action === 'assessmentPassword') {
-    var pl_pw_origUrl = req.cookies.pl_pw_origUrl;
-    var maxAge = 1000 * 60 * 60 * 12; // 12 hours
+  var redirectUrl = req.cookies.pl_pw_origUrl ?? '/';
+  var maxAge = 1000 * 60 * 60 * 12; // 12 hours
 
-    var pwCookie = csrf.generateToken({ password: req.body.password, maxAge }, config.secretKey);
-    res.cookie('pl_assessmentpw', pwCookie, { maxAge, httpOnly: true, secure: true });
-    res.clearCookie('pl_pw_origUrl');
-    return res.redirect(pl_pw_origUrl);
-  }
+  var pwCookie = csrf.generateToken({ password: req.body.password, maxAge }, config.secretKey);
+  res.cookie('pl_assessmentpw', pwCookie, { maxAge, httpOnly: true, secure: true });
+  res.clearCookie('pl_pw_origUrl');
+  return res.redirect(redirectUrl);
```

Comment:
It seems like this would be a really tragic thing to accidentally break, I assume we have some automated tests covering this. (this is one of the only times I've thought 'it would be nice to have a code coverage report' since we stopped generating them a year ago or whenever it was)

Reply:
Is there a specific concern you have with this diff?